lib/cc/service/http.rb

```diff
@@ -15,7 +15,41 @@ def default_http_options
     end
   end
 
-  def http_get(url = nil, params = nil, headers = nil)
+  RERAISE_ERRORS = [Test::Unit::AssertionFailedError]
+
+  def get(url = nil, body = nil, headers = nil, &block)
+    raw_get(url, body, headers, &block)
+  end
+
+  def post(url, body = nil, headers = nil, &block)
+    block ||= lambda{|*args| Hash.new }
+    response = raw_post(url, body, headers)
+    {
+      ok: response.success?,
+      params: body.as_json,
+      endpoint_url: url,
+      http_status: response.status
+    }.merge(block.call(response))
+  rescue CC::Service::HTTPError => e
+    {
+      ok: false,
+      params: body.as_json,
+      http_status: e.status,
+      endpoint_url: url,
+      exception_message: "#{e.class}: #{e.message}"
+    }
+  rescue *RERAISE_ERRORS
```

Comment:
This pattern is making me a little quesy-- can we remove this special casing for test errors and make the following rescue for `StandardError` more specific so it doesn't catch test assertion errors?